arch/arc/core/arc_smp.c

```diff
@@ -108,7 +108,7 @@ void arch_sched_ipi(void)
 	}
 }
 
-static int arc_smp_init(struct device *dev)
```

Comment:
Comment is on the commit titled:

    device: Const-ify all device driver instance pointers

which produced this change.  The provided Coccinelle script:

```
@r_const_dev_basic
  disable optional_qualifier
@
@@
-struct device *
+const struct device
```
is wrong in two ways:
* it changes a pointer to `struct device` to a `const struct device` object
* it fails to recognize a certain style that puts a `const` qualifier on the variable declaration.

For completeness it should be something like this:
```
@
disable optional_qualifier
@
@@
+const
 struct device *

@
disable optional_qualifier
@
@@
+const
 struct device * const
```

Reply:
sure